### DISCUSSION TITLE
Bafang DPC 181 can protocol open source firmware

### DISCUSSION THREAD

**Discussion by Gorky007** [danielnilsson9/bbs-fw#96] [Q&A]
Hello everyone. I am not a programmer and sorry for that and for my ignorance, but someone suggsted to me to ask on this platform if a sotfware/firmware has already been developed to make open source the Bafang DPC 181 canbus protocol with Bluetooth function. In particular, I would like to have the possibility to make it in communication with the Sram gx gearbox in order to preserve the transmission (since the Bafang M620 is a chain killer!), make it in communication with the Garmin edge 830, and perhaps to have also the possibility of displaying the messages and incoming phone calls from the cell phone. Many thanks in advance for Your help.

**Comment by danielnilsson9** [comment]
I don't think there has been any work. It's a cool idea though. It can probably be done as long as the chip in the display is not write protected but It's a significant amount of work to first reimplement the display firmware and then add additional function, not sure who would be up for that.
